File: matthews-app-backend/MatthewsApp.API/MatthewsApp.API/Models/Case.cs
using MatthewsApp.API.Enums;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Net.NetworkInformation;

namespace MatthewsApp.API.Models;

public class Case : BaseEntity
{

    [Column(TypeName = "nvarchar(256)")]
    [Required]
    public string ClientId { get; set; } // Promeni tip nakon što saznamo format ID-a

    [Column(TypeName = "nvarchar(256)")]
    [Required]
    public string ClientCaseId { get; set; }



    [Column(TypeName = "nvarchar(16)")]
    [Required]
    public string FirstName { get; set; }

    [Column(TypeName = "nvarchar(16)")]
    [Required]
    public string LastName { get; set; }

    public double Weight { get; set; }

    [Required]
    public GenderType Gender { get; set; }

    public ContainerType ContainerType { get; set; }

    public ContainerSize ContainerSize { get; set; }

    [Required]
    public bool IsObsolete { get; set; } = false;

    public int Age { get; set; }




    [Column(TypeName = "uniqueidentifier")]
    public Guid? ScheduledFacility { get; set; }

    [Column(TypeName = "uniqueidentifier")]
    public Guid? ScheduledDevice { get; set; }

    [Column(TypeName = "nvarchar(564)")]
    public string ScheduledDeviceAlias { get; set; }

    [Column(TypeName = "datetime2(7)")]
    public DateTime? ScheduledStartTime { get; set; }




    [Column(TypeName = "uniqueidentifier")]
    public Guid? ActualFacility { get; set; }

    [Column(TypeName = "uniqueidentifier")]
    public Guid? ActualDevice { get; set; }

    [Column(TypeName = "nvarchar(564)")]
    public string ActualDeviceAlias { get; set; }

    [Column(TypeName = "datetime2(7)")]
    public DateTime? ActualStartTime { get; set; }

    [Column(TypeName = "datetime2(7)")]
    public DateTime? ActualEndTime { get; set; }


    [Column(TypeName = "nvarchar(64)")]
    public string PerformedBy { get; set; }



    public string Fuel { get; set; }

    public string Electricity { get; set; }

    [Required]
    public Guid FacilityStatusId { get; set; }
    public FacilityStatus FacilityStatus { get; set; }

    [Column(TypeName = "nvarchar(256)")]
    public string? PhysicalId { get; set; }

    [NotMapped]
    public CaseStatus Status {
        get
        {
            return FacilityStatus.Status;
        }
    }
}

File: matthews-app/src/app/case/case.ts
/* eslint-disable @typescript-eslint/quotes */
import { GenderType } from "../core/enums";
import { GenderSelection } from "./selection-option";

/* eslint-disable @typescript-eslint/no-inferrable-types */
export class Case {
  id: string;

  clientId: string = '1'; // get ID from some API

  clientCaseId: string = '';

  firstName: string = '';

  lastName: string = '';

  weight: number;

  gender: number;

  genderText: string = '';

  containerType: number; // enum to be used instead

  containerTypeText: string = '';

  isObsolete: boolean = false;

  age: number;

  scheduledFacility: string = '00000000-0000-0000-0000-000000000000';

  scheduledDevice?: string = '00000000-0000-0000-0000-000000000000';

  scheduledDeviceAlias: string = '';

  scheduledStartTime?: string = '0001-01-01T00:00:00';

  actualFacility?: string = '00000000-0000-0000-0000-000000000000';

  actualDevice?: string = '00000000-0000-0000-0000-000000000000';

  actualDeviceAlias: string = '';

  actualStartTime?: string = '0001-01-01T00:00:00';

  actualEndTime?: string = '0001-01-01T00:00:00';

  createdBy: string = '00000000-0000-0000-0000-000000000000';

  createdTime: string = '0001-01-01T00:00:00';

  modifiedBy?: string = '00000000-0000-0000-0000-000000000000';

  modifiedTime?: string = '0001-01-01T00:00:00';

  performedBy?: string = '00000000-0000-0000-0000-000000000000';

  fuel: string = '';

  electricity: string = '';

  facilityStatusId: string = '00000000-0000-0000-0000-000000000000';

  facilityStatusText?: string = '';

  physicalId?: string = '';

  status: number = -1; 

  selected: boolean = false;

  //caseToFacilityStatuses: CaseToFacilityStatus[] = [];
}
